src/state/workspace/workspaceReducer.ts

```diff
@@ -101,6 +101,11 @@ const zoomReducer = (draftState: Draft<DiagramMakerWorkspace>, action: ZoomWorks
     newScale
   );
 
+  // Here, I should be able to check the current config.getViewMode
```

Comment:
Usually we cannot access the settings from reducers. We rely on action dispatchers to attach all the relevant actions on the action object itself and then reducers can remain simple & pure functions looking at the previous state & the action object only to produce the new state.

Can we use the view mode information in the action dispatcher to conditionally fire workspace pan action or workspace zoom action based on the view mode?